installer/Controller/MigrationController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Installer\Controller;
+
+class MigrationController implements Controller
+{
+	/**
+	 * @return array
+	 */
+	public function do()
+	{
+		$output = [];
+
+		$error = $this->migrate($output);
+		if (!$error) {
+			$error = $this->keyGenerate($output);
+		}
+		if (!$error) {
+			$this->installed($output);
+		}
+
+		return ['lines' => $output, 'errors' => $error];
+	}
+
+	/**
+	 * @return bool|null
+	 */
+	private function migrate(array &$output)
+	{
+		$res = 0;
+		exec('php artisan migrate', $output, $res);
```

Comment:
Can we not use `Artisan::call('migrate', ['--force' => true]);` for the initial migration?

Reply:
No, because `Artisan::call()` is not available yet. :)
This is bootstrapped, we do not load anything of Laravel yet.